### PR TITLE
Add Windows ProcessLauncher

### PR DESCRIPTION
**Pull request by pierricgimmig**
ProcessLauncher is a wrapper around CreateProcess and BusyLoopLauncher.
It is used to launch a process, optionally pausing it at the entry
point, and maintaining information required to suspend and resume the
process main thread.

### CODE REVIEW

File: src/WindowsUtils/ProcessLauncher.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
```
Comment (danielfenner) [inline src/WindowsUtils/ProcessLauncher.cpp:1]: 2022 :)

File: src/WindowsUtils/include/WindowsUtils/ProcessLauncher.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+// Class used to launch a process, optionally pausing it at the entry point and
+// maintaining information required to suspend and resume the process main thread.
+class ProcessLauncher {
+ public:
+  ErrorMessageOr<uint32_t> LaunchProcess(const std::filesystem::path& executable,
+                                         const std::filesystem::path& working_directory,
+                                         const std::string_view arguments,
+                                         bool pause_at_entry_point);
+
+  ErrorMessageOr<void> SuspendProcess(uint32_t process_id);
```
Comment (danielfenner) [inline src/WindowsUtils/include/WindowsUtils/ProcessLauncher.h:24]: The interface is a bit weird here: You can only suspend / resume processes started with `pause_at_entry_point==true`, right? And the Suspend / Resume need to be called in that sequence and can only be called once. Do you even need the `pause_at_entry_point==false` case here? 

I don't have a great idea on how to enforce the sequence of events (could be done by introducing classes for LaunchedProcess, SuspendedProcess, ResumedProcess but that seems to complicate things way more than it helps). 
So maybe just add a comments on the public members here and maybe similar in BusyLauncher where the same applies.

**Review by danielfenner — APPROVED** [review]
lgtm.

Please have a looks at the comment on the interface for `ProcessLauncher` - I think it is fine to just fix this with comments unless you have  a better idea.

File: src/WindowsUtils/include/WindowsUtils/ProcessLauncher.h (hunk 2 above)
Comment (pierricgimmig) [reply src/WindowsUtils/include/WindowsUtils/ProcessLauncher.h:27]: The sequence of events is enforced in the BusyLoopLauncher, which checks for the proper state transitions.

**Pull request merged**